src/main/java/org/folio/inventory/dataimport/handlers/matching/loaders/AbstractLoader.java

```diff
@@ -40,7 +40,7 @@ public LoadResult loadEntity(LoadQuery loadQuery, DataImportEventPayload eventPa
     loadResult.setEntityType(getEntityType().value());
     Context context = constructContext(eventPayload.getTenant(), eventPayload.getToken(), eventPayload.getOkapiUrl());
 
-    workerExecutor.executeBlocking(blockingFuture -> {
+    vertx.runOnContext(blockingFuture -> {
```

Comment:
Hi @RuslanLavrov, what is the difference between previous approach and the current one?

As far as I understand from the logs provided, the issue was caused because vert.x event bus thread was blocked by `future.join()` call.

Reply:
Hi @bohdan-suprun. The difference is to make the blocking method call `future.join()` from a thread of additional worker pool instead of threads, which serve inventory verticle.

Reply:
@RuslanLavrov 

> The difference is to make the blocking method call `future.join()` from a thread of additional worker pool instead of threads, which serve inventory verticle.

Does that mean that all of the threads in the dedicated pool for event processing could block instead of the shared pool?

Why does this operation need to be blocked at all? What is the join needed for?

Reply:
yes, threads from the additional pool will be used for running blocking method, it is needed
to return loadResult

Reply:
@RuslanLavrov 

> it is needed to return loadResult

Apologies, I really don't follow. Why is a blocking operation required to return a result? 

Why could the process not be invoked asynchronously? It is still possible to return a result at the end.

Reply:
Could this method return a CompletableFuture<LoadResult> instead of performing the join within it, then the client could decide how to continue?

This would avoid the need for a blocking call.